crates/aptos/src/move_tool/lint.rs

```diff
@@ -0,0 +1,65 @@
+// Copyright © Aptos Foundation
+// SPDX-License-Identifier: Apache-2.0
+
+use super::IncludedArtifactsArgs;
+use crate::common::types::{CliCommand, CliError, CliTypedResult, MovePackageDir};
+use aptos_framework::{BuildOptions, BuiltPackage};
+use async_trait::async_trait;
+use clap::Parser;
+use move_compiler_v2::Experiment;
+use move_model::metadata::CompilerVersion;
+
+/// Run a Lint tool to show additional warnings, in addition to ordinary warnings and/or errors
+/// generated by the Move 2 compiler.  Currently this is done by running the Move 2 compiler with a
+/// special pass, so some other compiler options are also available, but in the future this may be a
+/// dedicated tool.
+///
+/// Ideally, we would skip code generation, but currently this is not possible. We do omit
+/// metadata generation, which hopefully renders the output not able to be uploaded.
+#[derive(Parser)]
+pub struct LintPackage {
+    // TODO: add some options to select certain lint/warning passes?
+    #[clap(flatten)]
+    pub(crate) included_artifacts_args: IncludedArtifactsArgs,
+
+    #[clap(flatten)]
+    pub(crate) move_options: MovePackageDir,
+}
+
+#[async_trait]
+impl CliCommand<&'static str> for LintPackage {
+    fn command_name(&self) -> &'static str {
+        "LintPackage"
+    }
+
+    async fn execute(self) -> CliTypedResult<&'static str> {
+        let move_options = MovePackageDir {
+            move_2: true,
+            compiler_version: Some(CompilerVersion::V2_0),
+            ..self.move_options.clone()
+        };
+        if matches!(
+            self.move_options.compiler_version,
+            Some(CompilerVersion::V1)
+        ) {
+            return Err(CliError::CommandArgumentError(
+                "`aptos move lint` is not compatible with Move Compiler V1".to_string(),
+            ));
+        };
+        let more_experiments = vec![
+            Experiment::LINT_CHECKS.to_string(),
+            Experiment::SPEC_CHECK.to_string(),
+            Experiment::SEQS_IN_BINOPS_CHECK.to_string(),
+            Experiment::ACCESS_CHECK.to_string(),
+            Experiment::STOP_AFTER_EXTENDED_CHECKS.to_string(),
+        ];
+        let build_options = BuildOptions {
+            ..self
+                .included_artifacts_args
```

Comment:
Not sure what you're talking about here, but it's been changed.  PTAL.

Reply:
I think the line numbers messed up because of a possible rebase?

Anyway, I was referring to enabling of the experiment `SEQS_IN_BINOPS_CHECK` in `more_experiments`. You can discard this comment, as it is no longer applicable, because we *do* allow the lint with "move language v1", where as at the point when I reviewed this code, the language version was explicitly set to move v2.